host_wasmtime/src/main.rs

```diff
@@ -193,19 +198,16 @@ impl<T: WasiSensorView> wasi::sensor::sensor::HostDevice for T {
     ) -> Result<Result<Resource<wasi::sensor::sensor::Device>, wasi::sensor::sensor::DeviceError>>
     {
         trace!("opening a device {}", device_name);
-        let device_impl: Box<dyn SensorDevice + Send + Sync> = match &*device_name {
-            #[cfg(feature = "dummy")]
-            "dummy" => Box::new(match DummyDevice::new() {
-                Err(e) => return Ok(Err(e)),
-                Ok(i) => i,
-            }),
-            #[cfg(feature = "nokhwa")]
-            "nokhwa" => Box::new(match NokhwaDevice::new() {
-                Err(e) => return Ok(Err(e)),
-                Ok(i) => i,
-            }),
-            _ => return Ok(Err(wasi::sensor::sensor::DeviceError::NotFound)),
+        let v: Vec<&str> = device_name.split(":").collect();
+        if v.len() != 2 {
+            return Ok(Err(wasi::sensor::sensor::DeviceError::NotFound));
```

Comment:
"aaa:bbb"  would be mandatory ?

Reply:
in this particular implementation, yes.
in the wasi-sensor api, no.